web/src/main/java/com/epam/rft/atsy/web/controllers/NewApplicationController.java

```diff
@@ -56,7 +64,15 @@ public String saveOrUpdate(@Valid @ModelAttribute StateHistoryDTO stateHistoryDT
           .build();
 
       applicationsService.saveApplication(applicationDTO, stateHistoryDTO);
+      redirectAttributes.addAttribute("candidateId", stateHistoryDTO.getCandidateId());
+      return "redirect:/secure/candidate/details/{candidateId}";
     }
-    return "redirect:/secure/candidate/details/" + stateHistoryDTO.getCandidateId();
+
+    result.getFieldErrors().forEach(error ->
+        redirectAttributes.addFlashAttribute(error.getField() + "ErrorMessage",
+            messageKeyResolver.resolveMessageOrDefault(error.getDefaultMessage())));
+
+    redirectAttributes.addAttribute("candidateId", stateHistoryDTO.getCandidateId());
+    return "redirect:/secure/application";
```

Comment:
The mentioned redirect address is only valid when validation is OK. In case of a validation error user should be redirected to the "current" page where validation error messages are displayed.